code/game/gamemodes/abduction/machinery/console.dm

```diff
@@ -0,0 +1,184 @@
+	name = "Abductor console"
+	desc = "Ship command center."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "console"
+	density = 1
+	anchored = 1.0
+	var/obj/item/device/abductor/gizmo/gizmo
+	var/obj/item/clothing/suit/armor/abductor/vest/vest
+	var/obj/machinery/abductor/experiment/experiment
+	var/obj/machinery/abductor/pad/pad
+	var/obj/machinery/computer/camera_advanced/abductor/camera
+	var/list/datum/icon_snapshot/disguises = list()
+
+/obj/machinery/abductor/console/attack_hand(var/mob/user as mob)
+	if(..())
+		return
+	if(!IsAbductor(user))
+		user << "<span class='warning'>You start mashing alien buttons at random!</span>"
+		if(do_after(user,100))
+			TeleporterSend()
+		return
+	user.set_machine(src)
+	var/dat = ""
+	dat += "<H3> Abductsoft 3000 </H3>"
+
+	if(experiment != null)
+		var/points = experiment.points
+		dat += "Collected Samples : [points] <br>"
+	else
+		dat += "<span class='bad'>NO EXPERIMENT MACHINE DETECTED</span> <br>"
+
+	if(pad!=null)
+		dat += "<span class='warning'>Emergency Teleporter System.</span>"
+		dat += "<span clas=='warning'>Consider using primary observation console first.</span>"
+		dat += "<a href='?src=\ref[src];teleporter_send=1'>Activate Teleporter</A><br>"
+		dat += "<a href='?src=\ref[src];teleporter_set=1'>Set Teleporter</A><br>"
+		if(gizmo!=null && gizmo.marked!=null)
+			dat += "<a href='?src=\ref[src];teleporter_retrieve=1'>Retrieve Mark</A><br>"
+		else
+			dat += "<span class='linkOff'>Retrieve Mark</span><br>"
+	else
+		dat += "<span class='bad'>NO TELEPAD DETECTED</span></br>"
+
+	if(vest!=null)
+		dat += "<h4> Agent Vest Mode </h4><br>"
+		var/mode = vest.mode
+		if(mode == VEST_STEALTH)
+			dat += "<a href='?src=\ref[src];flip_vest=1'>Combat</A>"
+			dat += "<span class='linkOff'>Stealth</span>"
+		else
+			dat += "<span class='linkOff'>Combat</span>"
+			dat += "<a href='?src=\ref[src];flip_vest=1'>Stealth</A>"
+
+		dat+="<br>"
+		dat += "<a href='?src=\ref[src];select_disguise=1'>Select Agent Vest Disguise</a><br>"
+	else
+		dat += "<span class='bad'>NO AGENT VEST DETECTED</span>"
+	var/datum/browser/popup = new(user, "computer", "Abductor Console", 400, 500)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+	return
+
+/obj/machinery/abductor/console/Topic(href, href_list)
+	if(..())
+		return
+	if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+		usr.set_machine(src)
+		if(href_list["teleporter_set"])
+			TeleporterSet()
+		else if(href_list["teleporter_send"])
+			TeleporterSend()
+		else if(href_list["teleporter_retrieve"])
+			TeleporterRetrieve()
+		else if(href_list["flip_vest"])
+			FlipVest()
+		else if(href_list["select_disguise"])
+			SelectDisguise()
+		src.updateUsrDialog()
+
+/obj/machinery/abductor/console/proc/TeleporterSet()
+	var/A = null
+	A = input("Select area to teleport to", "Teleport", A) in teleportlocs
```

Comment:
After every input you need to recheck if they are able to still perform this action, as they can move away from the machine before choosing an answer.